pkg/apis/flowcontrol/types.go

```diff
@@ -265,6 +266,15 @@ type ResourcePolicyRule struct {
 	// +optional
 	// +listType=set
 	Namespaces []string
+
+	// `resourceNames` is a list of target resource names that restricts
+	// matches. A (get, update, patch, delete) request that specifies
+	// a target name matches if either (a) this list contains that target
+	// names or (b) this list contains "*".
+	// This list may be empty.
```

Comment:
I would not make empty permitted. I would make it default to `["*"]` instead.

Reply:
It looks like the rest of the PR does it the way I suggest, so it's actually just the comment that should be changed :)